### PR TITLE
add X509 request custom vectors

### PR DESCRIPTION
**Pull request by reaperhulk**
Verification of these vectors should be done by parsing the request and confirming it has the properties described.

### CODE REVIEW

File: docs/development/test-vectors.rst [hunk 1]
```diff
@@ -103,6 +103,15 @@ Custom X.509 Vectors
 * ``utf8_common_name.pem`` - An RSA 2048 bit self-signed CA certificate
   generated using OpenSSL that contains a UTF8String common name with the value
   "We heart UTF8!™".
+* ``dsa_sha1.pem`` - Contains a certificate request using 1024-bit DSA
+  parameters and SHA1 generated using OpenSSL.
+* ``rsa_md4.pem`` - Contains a certificate request using 2048 bit RSA and MD4 generated using OpenSSL.
```
Comment (alex) [inline docs/development/test-vectors.rst]: Line should be wrapped

**Comment by alex** [comment]
These are CSRs, not certificates, they need a new section in the docs IMO.

**Comment by reaperhulk** [comment]
Yeah I think I agree.

**Comment by public** [comment]
Verified :+1:

**Pull request merged**